python/ray/autoscaler/aws/config.py

```diff
@@ -229,7 +235,9 @@ def _configure_iam_role(config):
     assert profile is not None, "Failed to create instance profile"
 
     if not profile.roles:
-        role = _get_role(DEFAULT_RAY_IAM_ROLE, config)
+        role_name = CLOUDWATCH_RAY_IAM_ROLE if cwa_cfg_exists \
+            else DEFAULT_RAY_IAM_ROLE
```

Comment:
Ideally, I think we'll want to move this branch and all associated cloudwatch-centric logic (including instance profile names) into a separate cloudwatch_config.py file. We can then presumably use an AWS Config Provisioner factory to instantiate the appropriate aws config bootstrapper instance type at a higher-level, allowing us to consolidate any required branching in one location (e.g. within the provisioner that selects the appropriate config bootstrappe).